cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -132,7 +132,7 @@ class UploadResultToSw360Command : CliktCommand(
 
     private fun createSw360Release(pkg: Package, client: SW360ReleaseClientAdapter): SW360Release? {
         val sw360Release = SW360Release()
-            .setMainLicenseIds(pkg.declaredLicenses)
+            .setMainLicenseIds(pkg.declaredLicensesProcessed.spdxExpression?.licenses().orEmpty().toSortedSet())
```

Comment:
This omit unmappable licenses, which might a problem. I believe we should make use of `declaredLicensesProcessed.allLicenses` instead here, but then we'd still need to decide how to deal with unmapped licenses that contain spaces. What's SW360's approach here? Are spaces replaced e.g. with underscores or something like that?

Reply:
SW360 has its own model for licenses. I would assume that the license identifiers assigned to the releases must conform to this model. I do not know how it behaves when it encounters unknown license IDs. (Maybe we should ask this on the SW360 slack channel.) Licenses in SW360 have short names and normal names. The short names are probably rather IDs without whitespace or special characters.

The SW360 client library allows access to the licenses known to SW360; refer to [SW360LicenseClientAdapter](https://github.com/eclipse/sw360/blob/master/clients/client/src/main/java/org/eclipse/sw360/clients/adapter/SW360LicenseClientAdapter.java). So it would be possible to load the existing licenses to attempt a mapping, and even to create missing licenses in SW360 (which is of course more effort).

Reply:
Ok, thanks for this information. I indeed agree we should reach out to the SW360 folks to understand:

- What happens when passing a short name via `setMainLicenseIds()` that is unknown to SW360?
- Is there also a way to set the main license by normal name?
- What is the best way to deal with SPDX expression for license exceptions, like "GPL-2.0-or-later WITH Classpath-exception-2.0" as it contains spaces, but cannot be decomposed into multiple licenses, as "Classpath-exception-2.0" is no license on its own.

@mcjaeger or @JaideepPalit can you help here?

Reply:
So for now, I believe the best we can do is to slightly extend @bs-ondem's approach to also warn about unmapped licenses, like:

        // TODO: This omits operators and exceptions from licenses. We yet need to find a way to pass these to SW360.
        val licenseShortNames = pkg.declaredLicensesProcessed.spdxExpression?.licenses().orEmpty().toSortedSet()

        val unmappedLicenses = pkg.declaredLicensesProcessed.unmapped.toSortedSet()
        if (unmappedLicenses.isNotEmpty()) {
            log.warn {
                "The following licenses could not be mapped in order to create a SW360 release: $unmappedLicenses"
            }
        }

        val sw360Release = SW360Release()
            .setMainLicenseIds(licenseShortNames)
            .setName(pkg.id.name)
            .setVersion(pkg.id.version)

This at least solves the reported use-case of not being able to use "Apache License, Version 2.0" as a short license name. @bs-ondem can you please use the code from above?